testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/federation/ldap/LDAPUserMultipleCredentialTest.java

```diff
@@ -0,0 +1,173 @@
+
+/**
+ * Test user login with multiple credential providers, both local and federated through LDAP.
+ *
+ * @author Sophie Tauchert
+ */
+@EnableVault
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+public class LDAPUserMultipleCredentialTest extends AbstractLDAPTest {
+    @Page
+    protected LoginUsernameOnlyPage loginUsernameOnlyPage;
+    
+    @Page
+    protected PasswordPage passwordPage;
+    
+    @Page
+    protected SelectAuthenticatorPage selectAuthenticatorPage;
+    
+    @Page
+    protected LoginTotpPage loginTotpPage;
+    
+    @Rule
+    // Start an embedded LDAP server with configuration derived from test annotations before each test
+    public LDAPRule ldapRule = new LDAPRule()
+            .assumeTrue(LDAPTestConfiguration::isStartEmbeddedLdapServer);
+
+    @Override
+    protected LDAPRule getLDAPRule() {
+        return ldapRule;
+    }
+
+    @Override
+    protected void afterImportTestRealm() {
+        try {
+            getTestingClient().server().run(session -> {
+                LDAPTestContext ctx = LDAPTestContext.init(session);
+                RealmModel appRealm = ctx.getRealm();
+
+                // Delete all LDAP users
+                LDAPTestUtils.removeAllLDAPUsers(ctx.getLdapProvider(), appRealm);
+                // Add some new LDAP users for testing
+                LDAPObject user1 = LDAPTestUtils.addLDAPUser
+                        (
+                                ctx.getLdapProvider(),
+                                appRealm,
+                                "test-user",
+                                "John",
+                                "Doe",
+                                "test-user@something.org",
+                                "some street",
+                                "00000"
+                        );
+                LDAPTestUtils.updateLDAPPassword(ctx.getLdapProvider(), user1, "some-password");
+                LDAPObject user2 = LDAPTestUtils.addLDAPUser
+                        (
+                                ctx.getLdapProvider(),
+                                appRealm,
+                                "test-user-with-otp",
+                                "John",
+                                "Doe",
+                                "test-user-with-otp@something.org",
+                                "some street",
+                                "00000"
+                        );
+                LDAPTestUtils.updateLDAPPassword(ctx.getLdapProvider(), user2, "some-other-password");
+                UserModel userWithOtp = session.users().getUserByUsername(appRealm, "test-user-with-otp");
+                OTPCredentialModel otpCredential = OTPCredentialModel.createHOTP("DJmQfC73VGFhw7D4QJ8A", 6, 0, "HmacSHA1");
+                session.userCredentialManager().createCredential(appRealm, userWithOtp, otpCredential);
+            });
+        } catch (RunOnServerException ex) {
+            Assume.assumeFalse("Work around JDK-8214440",
```

Comment:
Could you please remove this catch block? I think it is not needed in most of the cases (EG. See LDAPProvidersIntegrationTest)

Reply:
Removed the try/catch block